src/libraries/Microsoft.PowerFx.LanguageServerProtocol/Protocol/BaseNLParams.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+namespace Microsoft.PowerFx.LanguageServerProtocol.Protocol
+{
+    public class BaseNLParams
+    {
+        /// <summary>
+        /// Additional context for NL operation. Usually, a stringified JSON object.
+        /// </summary>
+        public string NLContext { get; set; } = null;
```

Comment:
nit: Maybe just `Context` since it is already under NL specific requests? Dont' feel strongly about it though.